dts/bindings/reserved-memory/nordic,owned-memory.yaml

```diff
@@ -8,17 +8,64 @@ description: |
   will be recorded in the UICR of the compiled domain. Memory ownership and
   access is then configured for the domain at boot time, based on the UICR.
 
+  Example:
+
+    reserved-memory {
+        memory@2fc00000 {
+            compatible = "nordic,owned-memory";
+            reg = <0x2fc00000 0x1000>;
+            status = "okay";
+            nordic,access = <NRF_OWNER_ID_APPLICATION NRF_PERM_R>,
+                            <NRF_OWNER_ID_RADIOCORE NRF_PERM_W>;
+        };
+    };
+
+  A single local domain can request a memory region to be reserved on behalf of
+  multiple access owners. A single memory region shall be reserved by at most
```

Comment:
This rule would require a complex logic on the secure domain firmware. Let's discuss the following example planned in the memory map:

There are three concatenated RAM regions:

* Application's data
* Aplication - Radio shared IPC data
* Radio's data.

We have only two available MPC OVERRIDEs to describe these regions:

* Owner App, perm rw, adresses covering regions 1 and 2
* Owner Radio, perm rw, addresses covering regions 2 and 3

In your proposal build-time tools would need to verify that the three defined memory regions are concatenated in the memory. An algorithm would need to discover concatenated memory regions with the same owner and permissions and optimize number of allocated MPC OVERRIDEs.

Instead, I would suggest to make dts nodes representing memory configured in dts overrides like this:

```
app_rw_regions:
  nordic-access = application rw
  reg = <reg1start, size_of_regs_1and2>

radio_rw_regions:
  nordic-access = radio rw
  reg = <reg2start, size_of_regs_2and3>
```

Both nodes would includr sub-nodes describing nodes 1, 2; and 2, 3 accordingly.

This way mapping of the memory regions to MPC would be present explicilty in dts helping to design the memory layout properly and avoiding build time errors when a memory layout with implicit MPC mapping could not be translated to the available number of MPC entries.